src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplySum = 0;
+        int buySum = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] split = line.split(",");
+                String operation = split[0];
+                int amount = Integer.parseInt(split[1]);
+
+                if (operation.equals("supply")) {
+                    supplySum += amount;
+                }
+                if (operation.equals("buy")) {
+                    buySum += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error while reading file " + fromFileName, e);
+        }
 
+        int result = supplySum - buySum;
+
+        String report = "supply," + supplySum + System.lineSeparator()
```

Comment:
Use a StringBuilder to construct the report instead of concatenating strings. This is more efficient, especially since you are using concatenation in a loop.